ZenPacks/zenoss/OpenStackInfrastructure/modeler/plugins/zenoss/OpenStackInfrastructure.py

```diff
@@ -1604,11 +1604,24 @@ def preprocess_hosts(self, device, results):
         for service in results['cinder_services']:
             if 'host' in service:
 
-                # with the ceph backend, it seems like the cinder-volume
-                # hostname may not be relevant.  (At least on the test servers,
-                # I see 'ceph@ceph' here, which is meaningless)  (ZPS-3751)
-                if service['binary'] == 'cinder-volume' and service['host'] is not None and service['host'].endswith('@ceph'):
-                    LOG.debug("Ignoring host '%s' from cinder-volume service", service['host'])
+                # ------------------------------------------------------------------
+                # NOTE: (ZPS-3751) This is not the ultimate fix, but we need to
+                #       find some elegance before we continue. Perhaps a
+                #       zProperty that filters out these bad hosts will work.
+                #       Supressing arbitrary hosts is the ultimate goal.
+                #       This happens when we have:
+                #         1. Ceph Backend which create ceph@ceph hosts
+                #         2. hostgroups with TripleO which create 'hostgroup'
+                # ------------------------------------------------------------------
+                host_patterns_to_ignore = ['@ceph', 'hostgroup@']
+
+                if (service['host'] is not None
+                        and any([x for x in host_patterns_to_ignore if x in service['host']])
```

Comment:
I think these will need to be regexps if you want to do it this way- the old code had "endswith('@ceph')", but now you've changed it to a substring search, essentially.   So either have two checks, one with startswith and one with endswith, or have one check, using appropriate regexps.